diff --git a/dist/vue.runtime.esm.js b/dist/vue.runtime.esm.js
index 675cfeedc610be7aca9dbbe57a0d46139e3fc285..3e49a9943a59cabb77d48cc505c1695f5cca2223 100644
--- a/dist/vue.runtime.esm.js
+++ b/dist/vue.runtime.esm.js
@@ -5176,7 +5176,13 @@ function renderComponentRoot(instance) {
         handleError(err, instance, 1 /* ErrorCodes.RENDER_FUNCTION */);
         result = false;
     }
-    setRef$1(instance);
+    if (ctx.$mpPlatform === 'mp-xhs'){
+        setTimeout(() => {
+            setRef$1(instance);
+        });
+    } else {
+        setRef$1(instance);
+    }
     setCurrentRenderingInstance(prev);
     return result;
 }
@@ -5671,12 +5677,14 @@ function vOn(value, key) {
     const instance = getCurrentInstance();
     const ctx = instance.ctx;
     // 微信小程序，QQ小程序，当 setData diff 的时候，若事件不主动同步过去，会导致事件绑定不更新，（question/137217）
-    const extraKey = typeof key !== 'undefined' &&
-        (ctx.$mpPlatform === 'mp-weixin' || ctx.$mpPlatform === 'mp-qq') &&
-        (isString(key) || typeof key === 'number')
-        ? '_' + key
-        : '';
-    const name = 'e' + instance.$ei++ + extraKey;
+    let extraKey = typeof key !== "undefined" && (ctx.$mpPlatform === "mp-weixin" || ctx.$mpPlatform === "mp-qq") && (isString(key) || typeof key === "number") ? "_" + key : "";
+
+    const needExtraKey = ctx.$mpPlatform === 'mp-xhs' && ctx.$mpType === "component";
+    const eiCounter = instance.$ei++
+    if (needExtraKey) {
+        extraKey = "_" + ctx.componentId + "_" + eiCounter;
+    }
+    const name = "e" + eiCounter + extraKey;
     const mpInstance = ctx.$scope;
     if (!value) {
         // remove
